### PR TITLE
Updated File System API for Modern Browser Compatibility

### PR DESCRIPTION
**Pull request by Surya-palanisamy**
- Replaced the deprecated old File System API with the new API to ensure compatibility with modern browsers.
- Improved performance and security by leveraging the latest browser capabilities.
- Verified functionality across supported browsers for seamless file viewing.

### CODE REVIEW

File: client/src/components/sidebar/sidebar-views/FilesView.tsx [hunk 1]
```diff
@@ -62,6 +95,84 @@ function FilesView() {
         return children
     }
 
+    // Fallback method for browsers without File System Access API
+    const readFileList = async (files: FileList): Promise<FileSystemItem[]> => {
+        const children: FileSystemItem[] = []
+        const blackList = ["node_modules", ".git", ".vscode", ".next"]
+
+        for (let i = 0 i < files.length i++) {
```
Comment (sahilatahar) [inline client/src/components/sidebar/sidebar-views/FilesView.tsx:103]: Semicolon is missing after `let i = 0`

Comment (sahilatahar) [reply client/src/components/sidebar/sidebar-views/FilesView.tsx:103]: Make sure to check everything works perfectly.

**Comment by Surya-palanisamy** [comment]
Screenshot 2024-12-14 140015

## I've resolved the issue with the missing semicolon and verified that the implementation is now working properly. 
[link for preview](https://code-sync-pi.vercel.app/)

**Comment by sahilatahar** [comment]
Hi @Surya-palanisamy  Thank you for your contribution! This project already uses the modern **File System Access API** (`showDirectoryPicker`). However, you've added `webkitdirectory` and `webkitRelativePath`, which are older, non-standard APIs that can cause compatibility issues.  

Since the project prioritizes modern browser compatibility, it’s best to stick with `showDirectoryPicker`. Let me know if you’d like assistance refining the implementation!  

Thanks again for your effort!

**Comment by Surya-palanisamy** [comment]
@sahilatahar In this code-sync project, I am unable to upload files because the showDirectoryPicker API is not supported in Brave, Firefox, Could you please verify this issue?

**Comment by Surya-palanisamy** [comment]
### Thank you for your feedback! I’ve reviewed the browser support for webkitdirectory and webkitRelativePath and found that:

- These APIs are prefixed with webkit, meaning they are non-standard and not part of the official HTML5 specification.
- While they are supported in WebKit-based browsers like Chrome, Brave, and Safari and firefox (firefox not fully support)
- Due to their non-standard nature, they may only serve as a short-term solution and could lead to compatibility issues in the future.
- I understand the importance of using modern, standardized APIs like showDirectoryPicker for better browser support and long-term reliability. However, since showDirectoryPicker is not yet supported in Brave, Firefox, or Safari, this creates a challenge for consistent functionality across all browsers.

- Would you recommend implementing a fallback mechanism using webkitdirectory for now, while keeping showDirectoryPicker as the primary method for supported browsers? Let me know your thoughts on balancing compatibility and modern standards.

https://github.com/user-attachments/assets/f3f8251b-499c-437d-88eb-62462e83acc0

# video sample for brave and firefox with webkit
*Thank you again for your insights!*

**Comment by sahilatahar** [comment]
@Surya-palanisamy Okay, implement `showDirectoryPicker` as the primary method for supported browsers, and use `webkitdirectory` as a fallback for broader compatibility with older browsers. To check, use:

```javascript
if ('showDirectoryPicker' in window) {
    // Use showDirectoryPicker
} else if ('webkitdirectory' in HTMLInputElement.prototype) {
    // Fallback to webkitdirectory
}
```

**Comment by Surya-palanisamy** [comment]
@sahilatahar  Implemented showDirectoryPicker() as the primary method for directory selection. Added fallback using webkitdirectory for browsers that don't support the File System Access API.

**Comment by sahilatahar** [comment]
@Surya-palanisamy  Everything looks fine. Make sure to check browser compatibility and format code by using project's prettier configuration. There are semicolons on each line. I will merge it.

**Comment by Surya-palanisamy** [comment]
Done and test success 

https://github.com/user-attachments/assets/8aad0acc-6993-46c5-90bb-b0dc82a820f9

**Comment by Surya-palanisamy** [comment]
[live preview ](https://code-sync-pi.vercel.app) 

*Tested*

**Comment by Surya-palanisamy** [comment]
# done and sorted and formatted

[Final preview link](https://stream-share-code.vercel.app)

**Pull request closed without merging**